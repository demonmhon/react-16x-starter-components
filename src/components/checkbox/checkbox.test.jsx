import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Checkbox from './index';

describe('Checkbox component', () => {
  test('it matches the snapshot', () => {
    const component = create(<Checkbox />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('has the provided class name', () => {
    const className = 'the-custom-css-class';
    const component = shallow(<Checkbox className={className} />);
    expect(component.hasClass(className)).toBe(true);
  });

  describe('attributes', () => {
    test('default checked is false', () => {
      const component = shallow(<Checkbox />);
      expect(component.find('input').props().checked).toEqual(false);
    });

    test('checked added to component', () => {
      const props = {
        checked: true,
      };
      const component = shallow(<Checkbox {...props} />);
      expect(component.find('input').props().checked).toEqual(true);
    });

    test('disabled added to component', () => {
      const props = {
        disabled: true,
      };
      const component = shallow(<Checkbox {...props} />);
      expect(component.find('input').props().disabled).toEqual(true);
    });
  });
});
