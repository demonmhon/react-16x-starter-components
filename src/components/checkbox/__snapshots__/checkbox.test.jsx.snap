// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component it matches the snapshot 1`] = `
<div
  className="checkbox"
>
  <label
    className="checkbox__container"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="checkbox__ui"
    />
    <div
      className="checkbox__label"
    >
      
    </div>
  </label>
</div>
`;
